flink-connector-debezium/src/main/java/com/alibaba/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java

```diff
@@ -238,7 +238,7 @@ private long convertToLong(Object dbzObj, Schema schema) {
 
     private double convertToDouble(Object dbzObj, Schema schema) {
         if (dbzObj instanceof Float) {
-            return (double) dbzObj;
+            return ((Float) dbzObj).doubleValue();
```

Comment:
Could we add a test for this?  This is pretty interesting, you may meet precision problem.

Reply:
@leonardBang 

Thanks for the review.
Actually, debezium convert both MySQL `FLOAT` and `DOUBLE` type to `java.util.Double` with the schema `FLOAT64`. 
Please see [Debezium Documentation](https://debezium.io/documentation/reference/connectors/mysql.html).

The difference in precision we see is just because the float type is converted to the double type and It is no problem to restore it to float type again.
```java
float a = 123.102f;
double b = (double) a;

float c = (float) b;
```

So the code in condition one will not be executed.
```java
private double convertToDouble(Object dbzObj, Schema schema) {
    if (dbzObj instanceof Float) { //condition 1
         return (double) dbzObj;
    }
    ...
}
```

But purely semantically speaking, it will cause java.lang.ClassCastException: java.lang.Float cannot be cast to java.lang.Double.
```java
    public static void main(String[] args) {
        Float a = 123.102f;
        convertToDouble(a);
    }

    private static double convertToDouble(Object dbzObj) {
        if (dbzObj instanceof Float) {
            return (double) dbzObj;
        } else if (dbzObj instanceof Double) {
            return (double) dbzObj;
        } else {
            return Double.parseDouble(dbzObj.toString());
        }
    }
```

Reply:
Thanks @Jiabao-Sun for the detail explanation, the change make sense to me. 
User should use correcting data type mapping so that they may meet precision problem although our code is correct.